### DISCUSSION TITLE
How to get the direction a player is walking. (Not facing)

### DISCUSSION THREAD

**Discussion by DrMondGnu** [Minestom/Minestom#1197] [Help]
Is there an inbuild method to get the players **walking direction**? As i wan't to implement a dash whish should not go into the direction he is facing but rather the one he's walking. If there isn't a method for this, any ideas on how to calculate it?
Something like this:
`player.getPosition().getWalkingDirection();`

**Comment by MrGazdag** [comment]
If you mean the direction of the body of the player, then that is not visible serverside. That is only calculated clientside, and not tracked at all serverside.
You can test this, by having someone's body is looking like it is turning a bit to the right, then walking far enough from that player to unload them, then walking close again, and seeing that when they load, they are facing exactly forwards. You have to calculate this manually.

**Reply by DrMondGnu** [reply]
You are kinda right but i meant the direction the player is **walking** not the one the body is facing.

**Reply by MrGazdag** [reply]
Well you could listen to move events, and then calculate the difference. If you want rotation, you could use `Pos#withDirection(Point)`, to make a `Pos` point in that direction.
